src/checkovInstaller.ts

```diff
@@ -64,15 +168,40 @@ export interface CheckovInstallation {
     checkovInstallationMethod: CheckovInstallationMethod;
     checkovPath: string;
     workingDir?: string;
+    version?: string;
 }
 
 export const installOrUpdateCheckov = async (logger: Logger, installationDir: string): Promise<CheckovInstallation> => {
-    const dockerCheckovPath = await installOrUpdateCheckovWithDocker(logger);
-    if (dockerCheckovPath) return { checkovInstallationMethod: 'docker' , checkovPath: dockerCheckovPath };
-    const pip3CheckovPath = await installOrUpdateCheckovWithPip3(logger);
-    if (pip3CheckovPath) return { checkovInstallationMethod: 'pip3' , checkovPath: pip3CheckovPath };
-    const pipenvCheckovPath = await installOrUpdateCheckovWithPipenv(logger, installationDir);
-    if (pipenvCheckovPath) return { checkovInstallationMethod: 'pipenv' , checkovPath: pipenvCheckovPath, workingDir: installationDir };
-
-    throw new Error('Could not install Checkov.');
+    const autoUpdate = getAutoUpdate();
+    const checkovVersion = getCheckovVersion();
+
+    validateCheckovVersion(checkovVersion);
+
+    // when we attempt to install checkov, the method that succeeds determines the return value of this method.
+    // thus, we still need to run these methods, even if autoUpdate is false or there is a pinned checkov version,
+    // otherwise the plugin will not know which command to use
+    // the logic is a little different for each installer, so it's hard to reuse here
+
+    let installation: CheckovInstallation | null = null;
```

Comment:
Do you prefer this code better?
I'm more connected to immutable code (and functional if possible)

I feel like it's causing bloat that isn't adding in performance or readability (and is mutable which triggers me mixed feelings)

Reply:
I did this so we could log it at the end and call the version command.